doc/manual/faq.rst

```diff
@@ -1,110 +1,171 @@
----------------------------------------------------------------
+troubleshoot networking problems?
+---------------------------------
 
-  Like this::
+Diagnosis aids:
 
-    def run(self):
-        self.parse(self.pipe(iter(range(10))))
+    - Can you ``ping`` the device?
+    - Is the Ethernet LED on?
+    - Is the ERROR LED on?
+    - Is there anything unusual recorded in :ref:`the UART log <connecting-UART>`?
+
+Some things to consider:
+
+    - Is the ``core_addr`` field of your ``device_db.py`` set correctly?
+    - Are your core device's IP address and networking configurations definitely set correctly? Check the UART log for evidence of this, and ask your network administrator about the other settings.
+    - Is your core device configured for an external reference clock? If so, it cannot function correctly without one. Is the external reference clock plugged in?
+    - Are SFP0 and Ethernet plugged in all the way? Are they working? Try different cables and SFP adapters; M-Labs tests with CAT6 cables, but lower categories should be supported too.
+    - Are your PC and your crate in the same subnet?
+    - Is some other device in your network already using the configured IP address? Turn off the core device and try pinging the configured IP address; if it responds, you have a culprit. One of the two will need a different networking configuration.
+    - Are there restrictions or issues in your router or subnet that are preventing the core device from connecting? It may help to try connecting the core device to your PC directly.
+
+Command-line tools like ``nmap`` and ``arp`` can be used to scan your network, which may also help the core device be discovered.
+
+diagnose and fix sequence errors?
+---------------------------------
 
-    def pipe(self, gen):
-        for i in gen:
-            r = self.do(i)
-            yield r
+Go through your code, keeping manual track of SED lanes. See the following example: ::
 
-    def parse(self, gen):
-        for i in gen:
-            pass
+    @kernel
+    def run(self):
+        self.core.reset()
+        with parallel:
+            self.ttl0.on()  # lane0
+            self.ttl_sma.pulse(800*us)  # lane1(rising) lane1(falling)
+            with sequential:
+                self.ttl1.on()  # lane2
+                self.ttl2.on()  # lane3
+                self.ttl3.on()  # lane4
+                self.ttl4.on()  # lane5
+                delay(800*us)
+                self.ttl1.off() # lane5
+                self.ttl2.off() # lane6
+                self.ttl3.off() # lane7
+                self.ttl4.off() # lane0
+        self.ttl0.off()  # lane1 -> clashes with the falling edge of ttl_sma,
+                         # which is already at +800us
+
+In most cases, as in this one, it's relatively easy to rearrange the generation of events so that they will be better spread out across SED lanes without sacrificing actual functionality. One possible solution for the above sequence looks like: ::
```

Comment:
Maybe mention that unless events have to happen at **exactly** the same time, you can always just delay by a single RTIO period (8 ns on Kasli) between events to stay on the same lane.